src/Makefile.am

```diff
@@ -870,7 +872,7 @@ libbitcoinkernel_la_SOURCES = \
   index/base.cpp \
   index/blockfilterindex.cpp \
   index/coinstatsindex.cpp \
-  init/common.cpp \
+  kernel/init/common.cpp \
```

Comment:
Not sure if we want a `kernel` folder for code that is shared between node and kernel. An alternative would be to name it `init/kernel.cpp` or `init/globals.cpp`?

Otherwise it will be difficult to decide whether to put kernel code that is used elsewhere as well in `kernel` or in the "normal" hierarchy?  cc @ryanofsky

Reply:
I think the main thing here is to have a name that clearly indicates that this contains "only the common initialization code that the kernel needs" so that people don't accidentally add non-kernel code to it (it would result in a linker error in the libbitcoinkernel CI job anyway).

Other PRs in the project (e.g. #24410) does this as well, since "extracting only what the kernel needs" is the main goal of this phase of the project. I like using the `kernel/` prefix since it makes the aforementioned indication clear, and doesn't require inventing a new name.

In this particular case though, perhaps `kernel/init.cpp` would work.

One question is: do we keep the two functions moved in the `init::` namespace or the `kernel::` namespace.

In the longer run, I think eventually (very far off when `libbitcoinkernel` is very minimal) we could put everything that links into libbitcoinkernel into the `src/kernel/` folder.

Reply:
So is the goal to eventually split validation into kernel/validation.cpp and node/validation.cpp ? What about util functions that are included in the kernel?

Reply:
> So is the goal to eventually split validation into kernel/validation.cpp and node/validation.cpp ?

Yes, far off into the future we will likely do that (although I think in reality `node/validation.cpp` will likely be either: 1. a very very short file, or 2. deleted because what would be left belongs better in a differently named file).

> What about util functions that are included in the kernel?

I've discussed this with Ryan in #24352, and we came to the conclusion that we will move all the utils that `libbitcoinkernel` depends on to `libbitcoin_util`, now that's just w/re the libraries, we could do either one of two things for the hierarchy:

1. `src/kernel/` and `src/util/`, or
2. `src/kernel/` and `src/kernel/util/`

Either is fine with me. See here for more discussion: https://github.com/bitcoin/bitcoin/pull/24352#issuecomment-1048231861

Reply:
> so I am wondering if there should be two util libs? If yes, then it would make sense to have both `src/util` and `src/kernel/util`.  If not, then I don't care what is done.

I think that the point of https://github.com/bitcoin/bitcoin/pull/24352#issuecomment-1048231861 is to establish exactly what you're saying. In effect:

- `common` will be the non-kernel "util lib" (i.e. `src/util`), and
- `util` will be the kernel "util lib" (i.e. `src/kernel/util`).

Reply:
> I think that the point of https://github.com/bitcoin/bitcoin/pull/24352#issuecomment-1048231861 is to establish exactly what you're saying. In effect:

SGTM

> For example logging is used

Yes. Though, for the eventual libbitcoin kernel we need to find some way to do logging that doesn't make the client application depend on bitcoind's logging subsystem (say, a hook / notification registration). Out of scope here, of course.